propagator/opentelemetry-propagator-jaeger/src/opentelemetry/propagators/jaeger/__init__.py

```diff
@@ -79,9 +79,11 @@ def inject(
         span = trace.get_current_span(context=context)
         span_context = span.get_span_context()
         if span_context == trace.INVALID_SPAN_CONTEXT:
+            print(f'invalid span context')
             return
 
-        span_parent_id = span.parent.span_id if span.parent else 0
+        # NonRecordingSpans do not have a parent, skip injecting into them
+        span_parent_id = span.parent.span_id if getattr(span, 'parent', None) else 0
```

Comment:
Better to use [`hasattr`](https://docs.python.org/3/library/functions.html#hasattr) to check if an object has an attribute or not. Also, is the intention to `skip injecting` into the span if it does not have a parent? This is just setting `span_parent_id` to `0`.

Reply:
I was just trying to simplify the check. Originally I was skipping the injection but ended up just keeping the original behavior with a parent of `0`. I updated the comment.